Microsoft.HealthVault/Clients/Deserializers/ThingDeserializer.cs

```diff
@@ -19,11 +21,24 @@ namespace Microsoft.HealthVault.Clients.Deserializers
 {
     internal class ThingDeserializer : IThingDeserializer
     {
+        private readonly Guid _applicationSpecificId = new Guid("a5033c9d-08cf-4204-9bd3-cb412ce39fc0");
+
         private readonly IHealthVaultConnection _connection;
 
-        public ThingDeserializer(IHealthVaultConnection connection)
+        private readonly IThingTypeRegistrar _thingTypeRegistrar;
+
+        private readonly Dictionary<string, Dictionary<string, Type>> _appSpecificHandlers;
+        private readonly Dictionary<Guid, Type> _typeHandlers;
+        private readonly Dictionary<string, Type> _extensionHandlers;
+
+        public ThingDeserializer(IHealthVaultConnection connection, IThingTypeRegistrar thingTypeRegistrar)
         {
             _connection = connection;
+            _thingTypeRegistrar = thingTypeRegistrar;
+
+            _appSpecificHandlers = _thingTypeRegistrar.RegisteredAppSpecificHandlers;
+            _typeHandlers = _thingTypeRegistrar.RegisteredTypeHandlers;
+            _extensionHandlers = _thingTypeRegistrar.RegisteredExtensionHandlers;
```

Comment:
What happens when the user adds some type handlers after initializing the connection? The deserializer is a singleton so it will never pick up on any updates. I think we should either prevent the user from adding handlers after the first connection is resolved, or update this to always read from the IThingTypeRegistrar (and add some thread safety to ThingTypeRegistrarInternal)